user/pages/03.commerce2/02.developer-guide/09.promotions/02.create-a-condition/docs.md

```diff
@@ -4,4 +4,222 @@ taxonomy:
+A promotion condition evaluated only when a coupon is applied to an order.
+
+
+Minimum requirements to build a condition:
+---
+
+As you can see from comparing `ConditionInterface` with `ConditionBase`, 
+there is only one function required when extending ConditionBase to fulfill 
+the ConditionInterface prescription.
+
+```php
+<?php
+
+/**
+ * Defines the interface for conditions.
+ */
+interface ConditionInterface extends ConfigurablePluginInterface, PluginFormInterface, PluginInspectionInterface {
+
+  /** * Gets the condition label. */ 
+  public function getLabel();
+
+  /** * Shown in the condition UI when enabling/disabling a condition. */
+  public function getDisplayLabel();
+
+  /** * This is the entity type ID of the entity passed to evaluate(). */
+  public function getEntityTypeId();
+
+  /**
+   * Evaluates the condition.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity.
+   *
+   * @return bool
+   *   TRUE if the condition has been met, FALSE otherwise.
+   */
+  public function evaluate(EntityInterface $entity);
+}
+
+```
+[ConditionInterface.php](https://cgit.drupalcode.org/commerce/tree/src/Plugin/Commerce/Condition/ConditionInterface.php)
+
+```php
+<?php
+
+/**
+ * Provides the base class for conditions.
+ */
+abstract class ConditionBase extends PluginBase implements ConditionInterface {
+
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) { }
+
+  public function calculateDependencies() { ... }
+
+  public function defaultConfiguration() { ... }
+
+  public function getConfiguration() { ... }
+
+  public function setConfiguration(array $configuration) { ... }
+
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) { ... }
+
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {}
+
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) { ... }
+
+  // These are required by ConditionInterface.
+  public function getLabel() { ... }
+
+  public function getDisplayLabel() { ... }
+
+  public function getEntityTypeId() { ... }
+  
+  // TODO function evaluate is missing, write your own.
+}
+```
+[ConditionBase.php](https://cgit.drupalcode.org/commerce/tree/src/Plugin/Commerce/Condition/ConditionBase.php)
+
+
+Functionally, a condition would not be useful without some sort of user 
+configuration. defaultConfiguration, buildConfigurationForm, and 
+validateConfigurationForm go hand in hand with evaluate to create a basic 
+condition.
+
+Lets see how these functions are used in practice by analysing one of 
+the simplest conditions to ship with commerce: OrderItemQuantity.
+
+
+Example using class OrderItemQuantity:
+
+User story:
+  As a store manager I want to encourage bulk ordering by creating a promotion that applies to orders of at least 5 of the selected product. 
+  
+ ![Add Promotion UI](create-a-promotion.png)
+  
+
+```php
+<?php // namespace & use
```

Comment:
To be more concise you can drop this comment at the top. I think it's fair to assume that it would be there.